metrics-api/src/main/java/software/amazon/swage/metrics/MetricContext.java

```diff
@@ -53,9 +53,10 @@ default void record(Metric label, Number value, Unit unit) {
      * is often used to record occurrences of an event, such as the number of times
      * a method is called, an error occurred, or the amount of data sent.
      * <p/>
-     * Changes to the count are not timestamped as only the total value of all
-     * counts for a metric have any meaning. If the individual change needs to
-     * be tracked, it should be recorded as a gauged event.
+     * Changes to the count are timestamped with the time of context closure,
```

Comment:
This adds an explicit expectation/dependency on context closure to the context - now it assumes that every context will always be closed.  Is this appropriate?